src/azul/plugins/metadata/anvil/indexer/transform.py

```diff
@@ -400,7 +405,10 @@ def _file(self, file: EntityReference) -> MutableJSON:
                             uuid=file.entity_id)
 
     def _only_dataset(self) -> EntityReference:
-        return one(self._entities_by_type['dataset'])
+        try:
+            return one(self._entities_by_type['dataset'])
+        except ValueError:
```

Comment:
```suggestion
            return one(self._entities_by_type['dataset'], too_short=IndexError)
        except IndexError:
```
Could be worth adding just to re-enforce that the `except` is only handling the case of an empty iterable (and not if size is greater than one)